### DISCUSSION TITLE
Allow CORS: Access-Control-Allow-Origin

### DISCUSSION THREAD

**Discussion by felipe-mendieta** [infra-geo-ouverte/igo2#766] [Q&A]
when I modified the context file and added a wms service, it only works when I add the disable cors extension( CORS UNBLOCK) in my browser.
how do I avoid the cors error?

**Comment by pelord** [comment]
set crossOrigin: 'anonymous'

as in https://github.com/infra-geo-ouverte/igo2-lib/blob/4c06f2b90e1408c2d6fc77506d11264e20f4a7fe/packages/geo/src/lib/datasource/shared/datasources/wms-datasource.interface.ts#L14

and in 

https://github.com/infra-geo-ouverte/igo2-lib/blob/4c06f2b90e1408c2d6fc77506d11264e20f4a7fe/demo/src/app/geo/print/print.component.ts#L54

If this fix do not work. try to define your CORS in your webserver.

**Comment by pelord** [comment]
this is not specific to igo2-lib. it's a property from openlayers

https://openlayers.org/en/latest/apidoc/module-ol_source_ImageWMS-ImageWMS.html

**Comment by pelord** [comment]
@felipe-mendieta Is it working properly now with this fix?

**Reply by felipe-mendieta** [reply]
Where do I put crossOrigin: 'anonymous'? 
I try in src\contexts\peligro_volcanico.json
![image](https://user-images.githubusercontent.com/48095820/165635347-7ffaaadc-0ca0-4110-a4ae-6ffbee47ddad.png)

in this case as shown in the image does not work either.

When I use the proxy.conf.json it works but when I switch to production the proxy does not run anymore.

**Comment by pelord** [comment]
You must setup this  option into the source options...
You did right. 

The next step is to enable cors request into geoserver. 
https://docs.geoserver.org/latest/en/user/production/container.html#enable-cors . I don't know how to work with Geoserver...

**Comment by pelord** [comment]
@felipe-mendieta Ive tried on codesandbox. I get 1 main issue. Can you enable https over your server?

https://okbx42.csb.app/

**Comment by pelord** [comment]
![image](https://user-images.githubusercontent.com/7397743/167431967-ffcddaf6-66b7-4793-90a2-7b8a922e7259.png)

**Comment by pelord** [comment]
Once your server enabled with https, we could check if another web app can handle your server.

**Comment by pelord** [comment]
If you want a quick reply, we can talk into our gitter channel. 
https://gitter.im/igo2/Lobby

**Comment by pelord** [comment]
@felipe-mendieta Any updates?
